doc/extending/cop.txt

```diff
@@ -57,39 +57,63 @@ There are less methods to define for an Op than for a Type:
 
        *Default:* The default behavior is to do nothing.
 
-    .. method:: c_headers()
+    .. method:: c_headers([c_compiler])
 
        Returns a list of headers to include in the file. 'Python.h' is
        included by default so you don't need to specify it.  Also all
-       of the header required by the Types involved (inputs and
+       of the headers required by the Types involved (inputs and
        outputs) will also be included.
 
-    .. method:: c_header_dirs()
+       The `c_compiler` [#2v]_ parameter is the C compiler that will
+       be used to compile the code for the node. You may get multiple
+       calls with different C compilers.
+
+    .. method:: c_header_dirs([c_compiler])
 
        Returns a list of directories to search for headers (arguments
        to -I).
 
-    .. method:: c_libraries()
+       The `c_compiler` [#2v]_ parameter is the C compiler that will
+       be used to compile the code for the node. You may get multiple
+       calls with different C compilers.
+
+    .. method:: c_libraries([c_compiler])
 
        Returns a list of library names that your op needs to link to.
        All ops are automatically linked with 'python' and the
        libraries their types require. (arguments to -l)
 
-    .. method:: c_lib_dirs()
+       The `c_compiler` [#2v]_ parameter is the C compiler that will
+       be used to compile the code for the node. You may get multiple
+       calls with different C compilers.
```

Comment:
It makes a reference to the `[#2v]` "footnote" defined below.  This is because I didn't want to repeat myself in a lot of places so I move the common explanation there.